src/github.com/getlantern/balancer/balancer.go

```diff
@@ -49,7 +68,25 @@ func New(dialers ...*Dialer) *Balancer {
 // remaining Dialers until it either manages to connect, or runs out of dialers
 // in which case it returns an error.
 func (b *Balancer) DialQOS(network, addr string, targetQOS int) (net.Conn, error) {
-	dialers := b.dialers
+	var dialers []*dialer
+
+	// Checking destination port.
+	_, port, _ := net.SplitHostPort(addr)
+
+	// Are we attempting to connect to port 80 (plain HTTP)?
+	if port == "" || port == "80" {
+		// Then try to use only a trusted dialer.
+		dialers = b.trustedDialers()
+		// Unless we don't have any...
```

Comment:
I agree with you, this would be a very rare thing to happen. I personally would prefer having no connection at all in such cases but users may have a different opinion. I agree we should discuss this further.